fix: Add missing 'createdAt' property to CollectionComic type

Prompt for @claude:

Objective:
Fix the critical TypeScript error (TS2339) that is causing the Vercel deployment to fail.

Build Error:
src/pages/CollectionPage.tsx(63,32): error TS2339: Property 'createdAt' does not exist on type 'CollectionComic'.

Issue:
The sorting logic in CollectionPage.tsx attempts to sort the comic collection by a createdAt property. However, this property is not defined in the CollectionComic type interface, causing the TypeScript build to fail.

Required Fixes:

Locate the Type Definition: Find the file where the CollectionComic type is defined (likely in src/lib/types.ts or a similar location).

Update the Type: Add the createdAt property to the CollectionComic interface. It should be of type string or Date.

Verify Data Fetching: Ensure that the Supabase query in the data fetching service (e.g., collectionService.ts) is selecting the created_at column from the database. The property name in the database is likely created_at (snake_case) and should be mapped to createdAt (camelCase) in the application code.

Verification:
After these changes, the npm run build command must complete successfully with no TypeScript errors.

--- a/src/pages/CollectionPage.tsx
+++ b/src/pages/CollectionPage.tsx
@@ -60,7 +60,7 @@ const CollectionPage: React.FC = () => {
     purchasePriceValue: collectionComic.purchasePrice || 0,
     purchaseDate: collectionComic.purchaseDate ? new Date(collectionComic.purchaseDate).toLocaleDateString() : 'N/A',
     purchaseDateValue: collectionComic.purchaseDate ? new Date(collectionComic.purchaseDate) : null,
-    addedDate: collectionComic.createdAt ? new Date(collectionComic.createdAt) : new Date(),
+    addedDate: collectionComic.addedDate ? new Date(collectionComic.addedDate) : new Date(),
   })) || []
 
   // Create filtered and sorted comics using useMemo

